### PR TITLE
Add `TPESampler` for the tutorial paper

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

> [!TIP]
> Please follow the [Quick TODO list](https://github.com/optuna/optunahub-registry/tree/main) to smoothly merge your PR.

## Motivation

This sampler is to give a more stable implementation of [the TPE tutorial paper](https://arxiv.org/abs/2304.11127).

## Description of the changes

- Add a more parameterized `TPESampler` to reproduce the TPE tutorial paper

## TODO List towards PR Merge

- [x] Copy `./template/` to create your package
- [x] Replace `<COPYRIGHT HOLDER>` in `LICENSE` of your package with your name
- [x] Fill out `README.md` in your package
- [x] Add import statements of your function or class names to be used in `__init__.py`
- [x] Apply the formatter based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)
- [x] Check whether your module works as intended based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@eukaryo Could you check this PR?

**Comment by nabenabe0928** [comment]
I replaced the assignee with @y0z as @eukaryo is busy now.

**Comment by y0z** [comment]
@nabenabe0928 

Could you add an Abstract section to README.md? (cf. https://github.com/optuna/optunahub-registry/pull/130#event-13786300608)
It will help attract potential users to your package.

**Comment by nabenabe0928** [comment]
@y0z I added abstract!

File: package/samplers/tpe_tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def objective(trial):
+    x = trial.suggest_float("x", -5, 5)
+    y = trial.suggest_int("y", -5, 5)
+    z = trial.suggest_categorical("z", ["a", "aa", "aaa"])
+    return len(z) * (x**2 + y**2)
+
+
+module = optunahub.load_module(package="samplers/tpe_tutorial", repo_owner="nabenabe0928", ref="add-tpe-tutorial")
```
Comment (y0z) [inline package/samplers/tpe_tutorial/README.md:41]: The official package should be used when publishing.

```suggestion
module = optunahub.load_module(package="samplers/tpe_tutorial")
```

**Review by y0z — COMMENTED** [review]
I leave a comment.

**Comment by nabenabe0928** [comment]
@y0z done!

**Review by y0z — APPROVED** [review]
LGTM

**Pull request merged**